### PR TITLE
Remove the old AddDevice dialog and substitute it with the new wizard

### LINKED ISSUE #37616: Refresh "add new passkey" dialog
Currently, the "add new passkey" dialog prompts for a hardware key and shows an image that looks a lot like a YubiKey.

image

The dialog should be refreshed to match the new designs which make it clear that passkeys can be other types of authenticators, too:

image

See https://www.figma.com/file/8DQZl1mYUPLgvKgmA6zPOM/In-Progress-Work?type=design&node-id=7085-32060&mode=design&t=a3pMfHmh2ja3d87V-4 for designs.

### PR DESCRIPTION
**Pull request by bl-nero**
Fixes https://github.com/gravitational/teleport/issues/37616

Changelog: Updated the dialog for adding new authentication methods in the account settings screen.

### CODE REVIEW

File: web/packages/teleport/src/Account/Account.test.tsx [hunk 1]
```diff
@@ -47,7 +47,6 @@ describe('passkey + mfa button state', () => {
     ${'on'}       | ${false} | ${false}  | ${true}
     ${'optional'} | ${true}  | ${true}   | ${true}
     ${'optional'} | ${false} | ${false}  | ${true}
-    ${'otp'}      | ${true}  | ${false}  | ${true}
```
Comment (zmb3) [inline web/packages/teleport/src/Account/Account.test.tsx:50]: Why was this test case removed?

File: web/packages/teleport/src/Account/Account.tsx [hunk 2]
```diff
@@ -61,26 +58,8 @@ export default function AccountPage({ enterpriseComponent }: AccountPageProps) {
...
-
-  const canAddPasskeys =
-    cfg.isPasswordlessEnabled() &&
-    mfaOptions.some(option => option.value === 'webauthn');
-
-  const canAddMFA = mfaOptions.some(
-    option => option.value === 'otp' || option.value === 'webauthn'
-  );
+  const canAddPasskeys = cfg.isPasswordlessEnabled();
+  const canAddMFA = cfg.isMfaEnabled();
```
Comment (zmb3) [inline web/packages/teleport/src/Account/Account.tsx:62]:
```suggestion
  const canAddMfa = cfg.isMfaEnabled();
```

File: web/packages/teleport/src/Account/Account.tsx [hunk 3]
```diff
@@ -174,9 +150,13 @@ export function Account({
...
   }
 
-  function onAddPasskeySuccess() {
-    addNotification('info', 'Passkey successfully saved.');
-    onPasskeyAdded();
+  function onAddDeviceSuccess() {
+    const message =
+      newDeviceUsage === 'passwordless'
+        ? 'Passkey successfully saved.'
+        : 'MFA successfully saved.';
```
Comment (zmb3) [inline web/packages/teleport/src/Account/Account.tsx:157]:
```suggestion
        : 'MFA device successfully saved.';
```
Perhaps?

File: web/packages/teleport/src/Account/Account.test.tsx (hunk 1 above)
Comment (bl-nero) [reply web/packages/teleport/src/Account/Account.test.tsx:50]: @zmb3 It got broken when I removed a redundant check, but it's also impossible in real life: the OTP option [implies](https://github.com/gravitational/teleport/blob/bl-nero/new-auth-device-dialog-5/api/constants/constants.go#L221) passwordless being disabled. The auth server refuses to start if these options are in conflict. Similarly, we don't check what happens if 2fa is set to "off" and passwordless is set to true.

If you still think it's good to check these seemingly impossible conditions, I can bring back this case (and extend the "off" case) to verify that we always enable passwordless when the backend tells us so, regardless of the MFA setting itself.

Comment (bl-nero) [reply web/packages/teleport/src/Account/Account.test.tsx:50]: (On the second thought, I just went on and included these anyway.)

**Comment by bl-nero** [comment]
> The only issue I noticed is that I can't remove a passkey, even though I have other non-passwordless methods for logging in.

Filed https://github.com/gravitational/teleport/issues/38433 for it, as this is not something introduced by this PR.

File: web/packages/teleport/src/Account/Account.tsx [hunk 4]
```diff
@@ -61,32 +58,14 @@ export default function AccountPage({ enterpriseComponent }: AccountPageProps) {
...
-  );
+  const canAddPasskeys = cfg.isPasswordlessEnabled();
+  const canAddMfa = cfg.isMfaEnabled();
 
   return (
     <Account
       isSso={isSso}
       canAddPasskeys={canAddPasskeys}
-      canAddMFA={canAddMFA}
+      canAddMFA={canAddMfa}
```
Comment (gzdunek) [inline web/packages/teleport/src/Account/Account.tsx:68]: Nit: we can change it here as well:
```suggestion
     canAddMfa={canAddMfa}
```

**Review by gzdunek — APPROVED** [review]

**Review by kimlisa — APPROVED** [review]

**Pull request merged**